hazelcast/src/main/java/com/hazelcast/internal/partition/PartitionStamp.java

```diff
@@ -24,6 +24,8 @@
  */
 public final class PartitionStamp {
 
+    private static ThreadLocal<byte[]> BYTE_ARRAY = new ThreadLocal<>();
```

Comment:
`final`?

Reply:
BTW, should it be non-static and declared somewhere else? Different Hazelcast instance might have different partition count configured. Probably length check in `calculateStamp` might work.

Reply:
Got it, will check it out.

Reply:
Maybe we can have a thread-local `Int2ObjectHashMap` with the partition count as key, or just an array "indexed"  (envelop object with partition count and `byte[]`, linear search) with the partition count?

Reply:
Thanks guys. I opted for the length check and added a defensive zero-out.